helm-charts/basehub/values.yaml

```diff
@@ -369,3 +389,10 @@ jupyterhub:
           }
 
           c.KubeSpawner.environment.update(env)
+      07-2i2c-add-staff-user-ids-to-admin-users: |
+        from z2jh import get_config
+        add_staff_user_ids_to_admin_users = get_config("custom.2i2c.add_staff_user_ids_to_admin_users", False)
+        if add_staff_user_ids_to_admin_users:
+            user_id_type = get_config("custom.2i2c.add_staff_user_ids_of_type")
+            staff_user_ids = get_config(f"custom.2i2c.staff_{user_id_type}_ids", [])
+            c.JupyterHub.admin_users.extend(staff_user_ids)
```

Comment:
Most of the time, we need to add these to allowed_users as well i think - right? If not always, maybe we should have another boolean flag.

Reply:
We might be able to do something like:

```
authenticator_class = get_config("hub.config.JupyterHub.authenticator_class")
if authenticator_class == "github":
    # Unset allowed_users
else:
    c.Authenticator.extend(staff_user_ids)
```

Reply:
Since we explicitly set the authenticator class when using GitHub Teams auth

https://github.com/2i2c-org/infrastructure/blob/e28ec98cd6f61e410d1e25b966bd303cd84bff22/config/clusters/pangeo-hubs.cluster.yaml#L84

Reply:
It is always otherwise set to the following

https://github.com/2i2c-org/infrastructure/blob/e28ec98cd6f61e410d1e25b966bd303cd84bff22/deployer/hub.py#L459-L461

Reply:
Superb comment!